dataclients/kubernetes/kube.go

```diff
@@ -1400,13 +1400,19 @@ func setOriginMarker(s *clusterState, r []*eskip.Route) []*eskip.Route {
 		return nil
 	}
 
-	//it doesn't matter which route the marker is added to
-	r0 := r[0]
+	rr := make([]*eskip.Route, len(r))
+	copy(rr, r)
+
+	// it doesn't matter which route the marker is added to
+	// we also copy the route, to avoid storing it for the next diff comparison
```

Comment:
the goal is to store the routes with the data client without the origin markers, so that with the next comparison, there won't be any fake changes detected. There we append the markers to a copy.